### PR TITLE
Add type declarations for streams

### PR DESCRIPTION
**Pull request by bartlomieju**

### CODE REVIEW

**Comment by bartlomieju** [comment]
@Soremwar PTAL

File: node/_stream.d.ts [hunk 1]
```diff
@@ -0,0 +1,1500 @@
+// deno-lint-ignore-file no-explicit-any
+
+// Forked from https://github.com/DefinitelyTyped/DefinitelyTyped/blob/4f538975138678878fed5b2555c0672aa578ab7d/types/node/stream.d.ts
+
+import { EventEmitter } from "./_events.d.ts";
+import { Buffer } from "./buffer.ts";
+
+type BufferEncoding =
```
Comment (Soremwar) [inline node/_stream.d.ts]: A lesson I learned the hard way, it's better to just set BufferEncoding to a generic string and set its values as a comment block. There is a lot of code in the wild and even in std that do something like

```ts
function x(y: BufferEncoding){}
let z = "utf8";
x(z);
```
And that is not ok with TypeScript, so it's just best to let the validators in the functions do the job of checking that

**Review by Soremwar — COMMENTED** [review]
We need signatures for destroy, isDisturbed, and compose. However I don't see those in DefinitelyTyped so I think you gotta sintetize them from https://nodejs.org/api/stream.html#streamreadableisdisturbedstream

However, since this APIs seem to be experimental we can just skip them and add typings for them later on?

**Comment by Soremwar** [comment]
Signatures for `_isUint8Array` and `_uint8ArrayToBuffer` can be fixed by adding
```ts
export { _uint8ArrayToBuffer } from "./internal/streams/_utils.ts";
export { isUint8Array as _isUint8Array } from "./_util/_util_types.ts";
```

**Comment by bartlomieju** [comment]
> However, since this APIs seem to be experimental we can just skip them and add typings for them later on?

Sounds good.

File: node/_stream.d.ts (hunk 1 above)
Comment (bartlomieju) [reply node/_stream.d.ts]: Done

**Review by Soremwar — APPROVED** [review]
Thanks @bartlomieju

**Pull request merged**